src/table/cells/handleKeyPress.js

```diff
@@ -0,0 +1,74 @@
+export const moveToNextFocus = (rowElements, nextRow, nextCol) => {
+  const nextCell = rowElements[nextRow].getElementsByClassName('sn-table-cell')[nextCol];
+  nextCell.focus();
+  nextCell.setAttribute('tabIndex', '0');
+};
+
+export const arrowKeysNavigation = (evt, rowAndColumnCount, rowIndex, colIndex) => {
+  let nextRow = rowIndex;
+  let nextCol = colIndex;
+
+  switch (evt.key) {
+    case 'ArrowDown':
+      rowIndex + 1 < rowAndColumnCount.rowCount && nextRow++;
+      break;
+    case 'ArrowUp':
+      rowIndex > 0 && --nextRow;
+      break;
+    case 'ArrowRight':
+      colIndex < rowAndColumnCount.columnCount - 1 && nextCol++;
+      break;
+    case 'ArrowLeft':
+      colIndex > 0 && --nextCol;
+      break;
+    default:
+  }
+
+  return {
+    nextRow,
+    nextCol,
+  };
+};
+
+export const getRowAndColumnCount = (rootElement) => {
+  window.console.log(rootElement, 'rootElement');
+  const rowElements = rootElement.getElementsByClassName('sn-table-row');
+  const rowCount = rowElements.length;
+
+  const headCellElements = rootElement.getElementsByClassName('sn-table-head-cell');
+  const columnCount = headCellElements.length;
+
+  return { rowElements, rowCount, columnCount };
+};
+
+export const removeCurrentFocus = (evt) => {
+  evt.target.blur();
+  evt.target.setAttribute('tabIndex', '-1');
+};
+
+export const preventDefaultBehavior = (evt) => {
+  evt.stopPropagation();
+  evt.preventDefault();
+};
+
+const handleKeyPress = (evt, rootElement, rowIndex, colIndex) => {
```

Comment:
The simplest approach to testing this would be to mock all the methods on `evt` and `rootElement` then just make one test for one of the arrows and making sure everything is called (`preventDefault`, `target.blur` etc). This would automatically cover `preventDefaultBehavior`, `removeCurrentFocus` and `moveToNextFocus` as well.

Reply:
Thanks. Working on it.